airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/RouteToSyncTaskQueueActivityImpl.java

```diff
@@ -27,9 +29,13 @@ public RouteToSyncTaskQueueActivityImpl(final RouterService routerService) {
   public RouteToSyncTaskQueueOutput route(final RouteToSyncTaskQueueInput input) {
     ApmTraceUtils.addTagsToTrace(Map.of(CONNECTION_ID_KEY, input.getConnectionId()));
 
-    final String taskQueueForConnectionId = routerService.getTaskQueue(input.getConnectionId());
+    try {
+      final String taskQueueForConnectionId = routerService.getTaskQueue(input.getConnectionId());
 
-    return new RouteToSyncTaskQueueOutput(taskQueueForConnectionId);
+      return new RouteToSyncTaskQueueOutput(taskQueueForConnectionId);
+    } catch (final IOException e) {
```

Comment:
Now that the routerService queries the database, an IOException is possible. Throw a RetryableException if encountered

Reply:
Remind me again, where do we catch the RetryableException? I can't look since as my IntelliJ is pooping out on me.

Reply:
Asked here: https://airbytehq-team.slack.com/archives/C03AS1GAQV6/p1666217128557719

I think it extends `RuntimeException` which means Temporal knows to retry activities that fail with it as a cause, but it isn't super clear